modules/Attendance/attendance_take_byCourseClass.php

```diff
@@ -214,21 +213,31 @@
 
                         // Build the attendance log data per student
                         foreach ($students as $key => $student) {
-                            $data = array('gibbonPersonID' => $student['gibbonPersonID'], 'date' => $currentDate . '%');
+                            $data = array('gibbonPersonID' => $student['gibbonPersonID'], 'date' => $currentDate . '%', 'gibbonCourseClassID' => $gibbonCourseClassID);
                             $sql = "SELECT type, reason, comment, context, timestampTaken FROM gibbonAttendanceLogPerson
                                     JOIN gibbonPerson ON (gibbonAttendanceLogPerson.gibbonPersonID=gibbonPerson.gibbonPersonID)
                                     WHERE gibbonAttendanceLogPerson.gibbonPersonID=:gibbonPersonID
-                                    AND date LIKE :date";
-
-                            if ($prefillAttendanceType == 'N') {
-                                $data['gibbonCourseClassID'] = $gibbonCourseClassID;
-                                $sql .= " AND context='Class' AND gibbonCourseClassID=:gibbonCourseClassID";
-                            }
-                            $sql .= " ORDER BY timestampTaken DESC";
+                                    AND date LIKE :date
+                                    AND context='Class' AND gibbonCourseClassID=:gibbonCourseClassID
+                                    ORDER BY timestampTaken DESC";
                             $result = $pdo->executeQuery($data, $sql);
 
                             $log = ($result->rowCount() > 0) ? $result->fetch() : $defaults;
 
+                            //Check for school prefill if attendance not taken in this class
+                            if ($result->rowCount() == 0 ) {
+                                $data = array('gibbonPersonID' => $student['gibbonPersonID'], 'date' => $currentDate . '%');
+                                $sql = "SELECT type, reason, comment, context, timestampTaken FROM gibbonAttendanceLogPerson
+                                        JOIN gibbonPerson ON (gibbonAttendanceLogPerson.gibbonPersonID=gibbonPerson.gibbonPersonID)
+                                        WHERE gibbonAttendanceLogPerson.gibbonPersonID=:gibbonPersonID
+                                        AND date LIKE :date
+                                        AND NOT context='Class'
```

Comment:
My thinking was to avoid classes cross-filling each other, as it is less authoritative than school attendance. For example, a student might be in school but have skipped a lesson and be marked absent. The following class should not fill as Absent, otherwise the teacher might expect to not see them.